src/TaskLineRenderer.ts

```diff
@@ -31,6 +31,18 @@ export const LayoutClasses: { [c in TaskLayoutComponent]: string } = {
     blockLink: '',
 };
 
+const DataAttributeNames: { [c in TaskLayoutComponent]: string } = {
+    createdDate: 'taskCreated',
+    dueDate: 'taskDue',
+    startDate: 'taskStart',
+    scheduledDate: 'taskScheduled',
+    doneDate: 'taskDone',
+    priority: 'taskPriority',
+    description: '',
+    recurrenceRule: '',
+    blockLink: '',
+};
+
```

Comment:
Sorry but I feel this change makes the code more complicated to maintain rather than less.

This creates another non-obvious collection of values to populate in this file - and I am not seeing the advantage of it.

Why, for example, are some values empty? If I were to add a new type, how would I know what text value to add?

Reply:
> This creates another non-obvious collection of values to populate in this file - and I am not seeing the advantage of it.
> 
> Why, for example, are some values empty? If I were to add a new type, how would I know what text value to add?

Here is the code before the change:

```ts
switch (component) {
         case 'description':
         case 'recurrenceRule': {
             break;
         }
         case 'priority': {
             dataAttributes['taskPriority'] = PriorityTools.priorityNameUsingNormal(task.priority).toLocaleLowerCase();
             break;
         }
         case 'createdDate': {
             addDateDataAttributes(task.createdDate, 'taskCreated');
             break;
         }
         case 'dueDate': {
             addDateDataAttributes(task.dueDate, 'taskDue');
             break;
         }
         case 'startDate': {
             addDateDataAttributes(task.startDate, 'taskStart');
             break;
         }
         case 'scheduledDate': {
             addDateDataAttributes(task.scheduledDate, 'taskScheduled');
             break;
         }
         case 'doneDate': {
             addDateDataAttributes(task.doneDate, 'taskDone');
             break;
         }
```

1. `blockLink` option is empty - one could easily forget to add it. Now it is enforced by the compiler - if I comment out `blockLink` part, an error will pop up: `TS2741: Property 'blockLink' is missing in type '{ createdDate: string; dueDate: string; startDate: string; scheduledDate: string; doneDate: string; priority: string; description: string; recurrenceRule: string; }' but required in type '{ description: string; priority: string; recurrenceRule: string; createdDate: string; startDate: string; scheduledDate: string; dueDate: string; doneDate: string; blockLink: string; }'.`

So it it actually safer this way - the compiler will notify the user that an option of `TaskLayoutComponents` is missing.

2. `description` and `recurrenceRule` are also empty and do nothing

```ts
         case 'description':
         case 'recurrenceRule': {
             break;
         }
```

So the same questions shall be asked even without my change. Hence it is not the change provoking these questions. Which are totally valid btw.

Also please note that this is a step towards the abstraction of `TaskLayoutComponents`. This way there is more and more separation of data and behaviour.

Reply:
Although you have a point that data is scattered now so I included the dictionary in the function.

Reply:
> So it it actually safer this way - the compiler will notify the user that an option of `TaskLayoutComponents` is missing.

Interesting. I tested that out by removing a case from the switch block and still didn't get a warning.

So adding a value to `DataAttributeNames` gives seems to give the illusion of safety, but not an actual error message.

Reply:
Sorry, I was not clear, the error will appear if a value will be missing from the dictionary, not the switch-case.

Reply:
I understand that. I thought you were saying that before it was easy for someone to forget to add a value to the switch case...

So I tested it, and it's still possible for that to happen.

So I'm not understanding what extra safety I thought you were saying this change gives....

Anyway, it doesn't matter because I'm going to merge it after the meaning container is clarified, if you have time to add any comments...